### PR TITLE
Strip markdown formatting for platforms not supporting it

### PR DESCRIPTION
**Pull request by wm75**
an alternative to #69 

some advantages:
- less dependencies
- only one pass over html

### CODE REVIEW

**Comment by wm75** [comment]
doesn't handle blockquotes yet, but should be easy to add.

**Comment by wm75** [comment]
Well, that was more complicated than I thought, but blockquotes are now indicated via leading `>` just like in the original markdown.

**Comment by wm75** [comment]
Ready for review @bgruening @arash77

File: lib/plugins/linkedin.py [hunk 1]
```diff
@@ -47,16 +49,16 @@ def wrap_text_with_index(self, content):
...
-            [f"#{v}" for v in hashtags]
-        )  # the mentions are not linked to anyone!
+        hashtags = " ".join([f"#{v}" for v in hashtags])
         if len(images) > 20:
             warnings = f"A maximum of 20 images, not {len(images)}, can be included in a single linkedin post."
             images = images[:20]
         else:
             warnings = ""
 
+        content = strip_markdown_formatting(content)
```
Comment (arash77) [inline lib/plugins/linkedin.py:61]: Don't we need to split the text first like other platforms?
```suggestion
        content = "\n\n\n".join(
            strip_markdown_formatting(p) for p in content.split("\n\n\n")
        )
```

Comment (wm75) [reply lib/plugins/linkedin.py:61]: not really. The html parser in base.py now adds an extra line break before paragraphs.
Our readme currently says:

> You can also define custom breakpoints for splitting the post by inserting 2 empty lines anywhere in the text. The second empty line will simply be removed for media that don't support content splitting.

and this is exactly what will happen.

**Review by arash77 — APPROVED** [review]

**Comment by arash77** [comment]
Is this ok that it will always contain "\n\n" at the beginning?

**Comment by wm75** [comment]
> Is this ok that it will always contain "\n\n" at the beginning?

Good point. For plugins that do post stripping it doesn't matter since they are stripping surrounding whitespace from individual posts, but for linkedin that's an issue.

Will correct it.

**Comment by wm75** [comment]
Ah, another thing: the linkedin plugin *does* support post splitting, correct?
I didn't realize and then, of course, your earlier comment makes a lot of sense :sweat_smile: 

More fixes coming ...

File: lib/plugins/base.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            # add an empty line before paragraphs and headings
+            # do blockquoting as needed
+            if self.inside_blockquote:
+                if self.last_start_tag == "blockquote":
+                    p_sep = "\n\n> "
+                else:
+                    p_sep = "\n>\n> "
+            else:
+                p_sep = "\n\n"
+            self.text = self.text.rstrip("\n") + p_sep
```
Comment (arash77) [inline lib/plugins/base.py:44]: adding extra `\n` will also cause problem in situation like if the input contains something like this:
```markdown
- a

- b
```
the output will be this:
```

-  

a
- 

b


```

File: lib/plugins/base.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+# inspired by:
+# https://stackoverflow.com/questions/14694482/
+class HTMLFilter(HTMLParser):
+    text = ""
+    href = ""
+    li_count = 0
+    inside_blockquote = False
+    last_start_tag = ""
```
Comment (arash77) [inline lib/plugins/base.py:13]: Why not using `__init__` method here?
```suggestion
class HTMLFilter(HTMLParser):
    def __init__(self):
        super().__init__()
        self.text = ""
        self.href = ""
        self.li_count = 0
        self.inside_blockquote = False
        self.last_start_tag = ""
```

File: lib/plugins/base.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
+from html.parser import HTMLParser
+
+from markdown import markdown
```
Comment (arash77) [inline lib/plugins/base.py:3]:
```suggestion
from markdown2 import markdown
```

**Review by arash77 — COMMENTED** [review]
By using markdown2 we could detect the lists better.

File: lib/plugins/base.py (hunk 3 above)
Comment (wm75) [reply lib/plugins/base.py:13]: yes, would be cleaner, was just lazy :-)

File: lib/plugins/base.py (hunk 4 above)
Comment (wm75) [reply lib/plugins/base.py:3]: yes, but lets keep this change out of here and in a separate PR or issue where the pros and cons can be discussed properly.

File: lib/plugins/base.py (hunk 2 above)
Comment (wm75) [reply lib/plugins/base.py:44]: I suppose you've tested this?

Comment (wm75) [reply lib/plugins/base.py]: handling this now by turning line breaks on and off depending on context.

**Pull request merged**